bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAZoneHandler.java

```diff
@@ -0,0 +1,183 @@
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.measure.quantity.Temperature;
+
+import org.openhab.binding.plugwiseha.internal.PlugwiseHABindingConstants;
+import org.openhab.binding.plugwiseha.internal.api.exception.PlugwiseHAException;
+import org.openhab.binding.plugwiseha.internal.api.model.PlugwiseHAController;
+import org.openhab.binding.plugwiseha.internal.api.model.object.ActuatorFunctionalities;
+import org.openhab.binding.plugwiseha.internal.api.model.object.Location;
+import org.openhab.binding.plugwiseha.internal.config.PlugwiseHAThingConfig;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.plugwiseha.internal.handler.PlugwiseHABaseHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlugwiseHAZoneHandler} class is responsible for handling commands
+ * and status updates for the Plugwise Home Automation zones/locations.
+ * Extends @{link PlugwiseHABaseHandler}
+ *
+ * @author Bas van Wetten - Initial contribution
+ *
+ */
+@SuppressWarnings("unused")
+public class PlugwiseHAZoneHandler extends PlugwiseHABaseHandler<Location, PlugwiseHAThingConfig> {
+
+    // private PlugwiseHAThingConfig config = new PlugwiseHAThingConfig();
+    private @Nullable Location location;
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseHAZoneHandler.class);
+
+    // Constructor
+
+    public PlugwiseHAZoneHandler(Thing thing) {
+        super(thing);
+    }
+
+    public static boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return PlugwiseHABindingConstants.THING_TYPE_ZONE.equals(thingTypeUID);
+    }
+
+    // Overrides
+
+    @Override
+    protected synchronized void initialize(PlugwiseHAThingConfig config) {
+        if (thing.getStatus() == INITIALIZING) {
+            logger.debug("Initializing Plugwise Home Automation zone handler with config = {}", config);
+            if (!config.isValid()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                        "Invalid configuration for Plugwise Home Automation zone handler.");
+                return;
+            }
+
+            Optional.ofNullable(this.getPlugwiseHABridge()).ifPresent(bridge -> {
+                Optional.ofNullable(bridge.getController()).ifPresent(controller -> {
+                    this.location = getEntity(controller);
+
+                    setLocationProperties();
+                    updateStatus(ONLINE);
+                });
+            });
```

Comment:
I realize that you are doing this to deal with the null warnings but I would much rather you cache the field in a local variable than wrap the field in an Optional every time you want to access it.